solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -304,9 +328,9 @@ public List<SearchComponent> getComponents() {
     return result;
   }
 
-  private boolean isDistrib(SolrQueryRequest req) {
-    boolean isZkAware = req.getCoreContainer().isZooKeeperAware();
-    boolean isDistrib = req.getParams().getBool(DISTRIB, isZkAware);
+  protected boolean isDistrib(SolrQueryRequest req, ResponseBuilder rb) {
+    boolean theDefault = req.getCoreContainer().isZooKeeperAware() || rb.isForcedDistrib();
```

Comment:
This change allows a component using `setForcedDistrib` to _also_ be a condition for distrib becoming true, which is relevant in standalone.  But as I look at this again... I now question the decision.  Arguably, it's nicer to keep the story of standalone consistent to be `distrib` is by default `false` in standalone mode and not in SolrCloud.  No exceptions.  Such components which can only work in a distributed mode could throw an exception 400, saying distrib must be enabled (since that's how standalone mode works).  To better support that, isDistrib could be initialized first thing in handleRequestBody.  WDYT?